federation/pkg/federation-controller/service/dns.go

```diff
@@ -89,23 +90,40 @@ func (s *ServiceController) getClusterZoneNames(clusterName string) (zones []str
 	return client.cluster.Status.Zones, client.cluster.Status.Region, nil
 }
 
-// getFederationDNSZoneName returns the name of the managed DNS Zone configured for this federation
-func (s *ServiceController) getFederationDNSZoneName() (string, error) {
-	return s.zoneName, nil
+// getFederationDnsSuffix returns the DNS suffix to use for records for this federation
+func (s *ServiceController) getFederationDnsSuffix() (string, error) {
+	return s.federationDnsSuffix, nil
 }
 
-// getDnsZone is a hack around the fact that dnsprovider does not yet support a Get() method, only a List() method.  TODO:  Fix that.
-func getDnsZone(dnsZoneName string, dnsZonesInterface dnsprovider.Zones) (dnsprovider.Zone, error) {
-	dnsZones, err := dnsZonesInterface.List()
+// getDnsZone returns the zone, as identified by zoneName and/or zoneID
+func (s *ServiceController) getDnsZone() (dnsprovider.Zone, error) {
```

Comment:
nit: As per similar comment on another PR, I don't really like this becoming a member function of ServiceController.  I prefer the original decoupling.  It originated as a simple utility function to get around the fact that dnsprovider does not support Get().  I think it should stay that way. But I don't feel strongly enough about it to fight :-)